### PR TITLE
P0019R8 Atomic ref

### LINKED ISSUE #2141: [2018-06 LWG Motion 24] P0019R8 Atomic ref

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #2141.

### CODE REVIEW

File: source/atomics.tex [hunk 1]
```diff
@@ -1503,6 +1508,723 @@
...
+\pnum
+\returns Atomically returns the value referenced by \tcode{*ptr}.
+\end{itemdescr}
+
+\indexlibrarymember{exchange}{atomic_ref}%
+\indexlibrarymember{exchange}{atomic_ref<T*>}%
+\indexlibrarymember{exchange}{atomic_ref<\placeholder{integral}>}%
+\indexlibrarymember{exchange}{atomic_ref<\placeholder{floating-point}>}%
+\begin{itemdecl}
+T exchange(T desired, memory_order order = memory_order_seq_cst) noexcept;
```
Comment (timsong-cpp) [inline source/atomics.tex]: The paper is missing a `const` here, presumably a typo (since it's in the class definition).

Comment (jensmaurer) [reply source/atomics.tex]: Fixed.

File: source/atomics.tex [hunk 2]
```diff
@@ -1503,6 +1508,723 @@
...
+\end{note}
+
+\rSec2[atomics.ref.operations]{Operations}
+
+\indexlibrarymember{is_always_lock_free}{atomic_ref}%
+\indexlibrarymember{is_always_lock_free}{atomic_ref<T*>}%
+\indexlibrarymember{is_always_lock_free}{atomic_ref<\placeholder{integral}>}%
+\indexlibrarymember{is_always_lock_free}{atomic_ref<\placeholder{floating-point}>}%
+\begin{itemdecl}
+static constexpr bool is_always_lockfree;
```
Comment (timsong-cpp) [inline source/atomics.tex]: This is misspelled in the paper.

Comment (jfbastien) [reply source/atomics.tex]: Yeah should be `is_always_lock_free`

Comment (jensmaurer) [reply source/atomics.tex]: Fixed.

**Pull request merged**